### PR TITLE
Add javadsl FormData.create(Iterable) overload #21303

### PR DESCRIPTION
**Pull request by matsluni**
- Added FormData.create overload.
- Didn't know where to put a test, but would be happy to add one.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by johanandren** [comment]
A test would go in `src/test/java/akka/http/javadsl/model` or possibly, if you want to write the testcase in scala in ``src/test/scala/akka/http/javadsl/model`. We do not have much unit tests for these simple model factory methods in general, but that doesn't mean it's a bad idea!

File: akka-http-core/src/main/java/akka/http/javadsl/model/FormData.java [hunk 1]
```diff
@@ -51,4 +53,14 @@ public static FormData create(Pair<String, String>... params) {
...
     return new FormData(Query.create(params));
   }
+
+  /**
+   * Creates a FormData from the given parameters.
+   */
+  public static FormData create(Iterable<Pair<String, String>> params) {
+    @SuppressWarnings("unchecked")
+    final Pair<String, String>[] paramsArray =
+          StreamSupport.stream(params.spliterator(), false).toArray(Pair[]::new);
```
Comment (johanandren) [inline akka-http-core/src/main/java/akka/http/javadsl/model/FormData.java]: Isn't it a bit expensive to kick of a Java stream just to collect the iterable into an array?

Comment (johanandren) [inline akka-http-core/src/main/java/akka/http/javadsl/model/FormData.java]: Can't it just call `Query.create(Iterable<Pair<...>>)` instead?

Comment (matsluni) [inline akka-http-core/src/main/java/akka/http/javadsl/model/FormData.java]: Totally correct! Will do this! I just copied from Query.create(Iterable...), but I can call this instead.

**Comment by matsluni** [comment]
I updated my commit wrt your comments. Due to a lack of time I leave out the tests for now.

**Comment by johanandren** [comment]
LGTM, thanks!

**Comment by ktoso** [comment]
Too bad that missing test. I'll merge and move it to the right repository, thanks again @matsluni

**Comment by ktoso** [comment]
Resolves #21303

**Pull request merged**